---
title: "Ca. Methanobrevibacter cohabitans (pr_cl_1990) Tree and Metadata"
author: "Keri Burge"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("ggtree")


install.packages("aplot")
```


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(janitor)
library(ape)
library(ggtree)
library(aplot)
library(viridis)
library(tidyverse)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# Colors
```{r}
# Species clusters (5)
cluster_colors <- c("1188" = "#4292C6",
                      "1189" = "#A1D99B",
                      "1187" = "#FCB076",
                      "1190" = "#E794C1",
                      "Reference" = "lightgrey")


continent_colors <- c("Africa" = "#CC79A7",
                      "Asia" = "#97CE2F",
                      "Caribbean" = "#F09163",
                      "Europe" = "#9E9AC8",
                      "North America" = "#1ec1a7",
                      "Oceania" = "#098BD9",
                      "Reference" = "lightgrey")

mag_source_colors <- c("Ancient Human" = "#FFD5AF",
                       "Neanderthal" = "#FEEDA0",
                       "Modern non-industrial" = "#BCE1FF",
                       "Modern industrial" = "#EFB6D6",
                       "Non-human primate" = "#b2815e",
                       "Reference" = "lightgrey")

crispr_colors <- c("III-A" = "#4590fe",
             "I-B" = "#E794C1",
             "IV-B" = "#8D5D4B",
             "II-D" = "black",
             "None" = "#f0f0f0")

cazyme_colors <- c("Cell wall processing" = "#BCBDDC")

prophage_colors <- c("Present" = "#f6865e",
              "Absent" = "#FBD9D2")

methanogenesis_colors <- c("Methanogenesis" = "#A3E4D7",
                           "F420 biosynthesis" = "#FEC44F")


fastBAPS_clusters_colors <- c("1" = "#9ECAE1",
                              "2" = "#b2df8a",
                              "3" = "#BCBCD4",
                              "4" = "#E2BED0",
                              "5" = "#FAC3B6",
                              "6" = "#D5C1B6",
                              "7" = "#FFFD78",
                              "8" = "#32a02d",
                              "9" = "#2078b4")

```

# Metadata
```{r}
metadata <- fread("./05-results/methanobrevibacter_metadata.tsv") %>%
  filter(primary_cluster == 1190) %>%
  rename(Label = Bin) %>%
  rename(Bin = prepublication_bin) %>%
  relocate(Bin, .before = Label)
  
```

# Complete tree
## Root-finding with MAD 
Genome Biol Evol. 2012 May 16;4(8):821â€“831. doi: 10.1093/gbe/evs047
```{r}
source("02-scripts/mad.R")

```

```{r}
tre_methanobrevibacter_c1190_scc <- ape::read.tree("05-results/methanobrevibacter_c1190_scc.raxml.support")

methanobrevibacter_c1190_mad_root <- mad(tre_methanobrevibacter_c1190_scc, 'newick')

methanobrevibacter_c1190_mad_root %>%
  as.list() %>%
  fwrite(., "./04-analysis/phylogenies/methanobrevibacter/cluster_4_w_CSC/scc_non_recomb/methanobrevibacter_c4_mad_root.nwk", quote = F)
```

```{r}
#Find the root
methanobrevibacter_c1190_mad_root <- mad(tre_methanobrevibacter_c1190_scc, 'full')

#ERSX0005001-megahit_029
```


```{r}
methanobrevibacter_c1190_mad_root <- ape::read.tree("./04-analysis/phylogenies/methanobrevibacter/cluster_4_w_CSC/scc_non_recomb/methanobrevibacter_c4_mad_root.nwk")

methanobrevibacter_c1190_mad_root <- root(methanobrevibacter_c1190_mad_root, outgroup = "ERSX0005001-megahit_029")

methanobrevibacter_c1190_mad_root$tip.label <- methanobrevibacter_c1190_mad_root$tip.label %>%
  as.data.frame() %>%
  dplyr::rename(Bin = 1) %>%
  mutate(Bin = if_else(str_detect(Bin, "GC|refbinned"), Bin, 
                        str_replace_all(Bin, "-metaspades_00", ".bin.r.") %>%
                        str_replace_all("-metaspades_0", ".bin.r.") %>%
                        str_replace_all("-metaspades", ".bin.r.") %>%
                        str_replace_all("-megahit_00", ".bin.r.") %>%
                        str_replace_all("-megahit_0", ".bin.r.") %>%
                        str_replace_all("-megahit_", ".bin.r.") %>%
                        str_replace_all("_", ""))) %>%
  pull(Bin)


# create tree
methanobrevibacter_tree_c1190 <- ggtree(methanobrevibacter_c1190_mad_root, ladderize = T)

host_shapes <- c("Homo sapiens" = 16, "Homo sapiens neanderthalensis"  = 15, "Pan troglodytes schweinfurthii" = 17, "Reference" = 12, "Papio hamadryas" = 18)

methanobrevibacter_tree_c1190_age <- methanobrevibacter_tree_c1190 %<+% metadata + 
  geom_tippoint(aes(color = log_age, shape = sample_host), size = 3.5) +
  scale_size_continuous(limits = c(0,3)) +
  theme(legend.position="right") + 
  #scale_color_viridis_c(option = "inferno", name = "Age (BP) (log10)", begin = 0.2, end = 1) +
  scale_color_viridis_c(option = "mako", name = "Age (BP) (log10)", direction = -1, limits = c(0.845098, 5.008600)) +
  scale_shape_manual(values = host_shapes, name = "Sample Host") +
  geom_treescale(x = 0, y = 5, width = 0.005, fontsize = 4) +
  geom_nodelab(aes(label = ifelse(as.numeric(label) > 50, label, "")), size = 2, hjust = -0.1) +
  guides(shape = guide_legend(override.aes = list(size = 3))) +
  geom_tiplab(aes(label = Label), size = 3, hjust = -0.1)

methanobrevibacter_tree_c1190_age

```


## Species cluster, MAG source, Continent
```{r}
#fastBAPS cluster
fastBAPS <- fread("05-results/fastBAPS_methanobrevibacter_cluster4_snp_counts_and_clusters.tsv") %>%
  mutate(Bin = if_else(str_detect(Bin, "GC|refbinned"), Bin, 
                        str_replace_all(Bin, "-metaspades_00", ".bin.r.") %>%
                        str_replace_all("-metaspades_0", ".bin.r.") %>%
                        str_replace_all("-metaspades", ".bin.r.") %>%
                        str_replace_all("-megahit_00", ".bin.r.") %>%
                        str_replace_all("-megahit_0", ".bin.r.") %>%
                        str_replace_all("-megahit_", ".bin.r.") %>%
                        str_replace_all("_", ""))) %>%
  mutate(Cluster = as.character(Cluster))

species_clusters_heatmap <- ggplot(fastBAPS, aes(x = "Cluster", y = Bin, fill = Cluster)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_manual(values = fastBAPS_clusters_colors, name = "fastBAPS cluster")

#Continent
continent_heatmap <- ggplot(metadata, aes(x = "Continent", y = Bin, fill = Continent)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_manual(values = continent_colors, name = "Continent")

#MAG Source
mag_source_heatmap <- metadata %>%
  mutate(Industrialization = if_else(Industrialization == "", "Reference", Industrialization)) %>%
  mutate(Industrialization = ifelse(str_detect(sample_host, "Homo sapiens neanderthalensis"), "Neanderthal", Industrialization)) %>%
  mutate(Industrialization = str_replace_all(Industrialization, "Non_industrial", "Modern non-industrial"),
         Industrialization = str_replace_all(Industrialization, "Industrial", "Modern industrial"),
         Industrialization = str_replace_all(Industrialization, "Ancient", "Ancient Human")) %>%
  ggplot(., aes(x = "Industrialization", y = Bin, fill = Industrialization)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_manual(values = mag_source_colors, name = "MAG Source")
  
```



## Additional metada
### Age
```{r}
heatmap_age <- ggplot(metadata, aes(x = "log_age", y = Bin, fill = log_age)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_viridis_c(option = "mako", name = "Age (BP) (log10)", direction = -1, limits = c(0.845097, 5.008601))

heatmap_age
```


### Latitude
```{r}
methanobrevibacter_coords <- fread("00-documentation/methanobrevibacter_sample_coords.tsv") %>%
  inner_join(., metadata, by = "Bin")

latitude <- ggplot(methanobrevibacter_coords, aes(x = "latitude", y = Bin, fill = latitude)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_gradient2(low =  "#331068", mid = "#FCFDBF", high = "#E95562", midpoint = 0, name = "Latitude", limits = c(-34.03000, 63.43100))

latitude 
```


### Longitude
```{r}
longitude <- ggplot(methanobrevibacter_coords, aes(x = "longitude", y = Bin, fill = longitude)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_gradient2(low =  "#331068", mid = "#FCFDBF", high = "#E95562", midpoint = 0, name = "Longitude", limits = c(-151.400000, 177.500000))

longitude 
```


### Contig Count
```{r}
methanobrevibacter_stats <- fread("00-documentation/methanobrevibacter_mag_stats.tsv") %>%
  inner_join(., metadata)

contigs <- ggplot(methanobrevibacter_stats, aes(x = "quast_n_contigs", y = Bin, fill = quast_n_contigs)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_viridis_c(option = "A", direction = -1, name = "Contig count", limits = c(67, 1233))

contigs
```


### N50
```{r}
n50 <- methanobrevibacter_stats %>%
  select(Bin, quast_N50) %>%
  mutate(n50 = ifelse(quast_N50 <= 5000, "5000",
                      ifelse(quast_N50 > 5000 & quast_N50 <= 10000, "10000",
                             ifelse(quast_N50 > 10000 & quast_N50 <= 50000, "50000",
                                    ifelse(quast_N50 > 50000 & quast_N50 <= 100000, "100000","150000"))))) %>%
  dplyr::select(-quast_N50) %>%
  mutate(n50 = fct_relevel(n50, "5000","10000","50000","100000")) %>%
  mutate(n50 = factor(n50, levels = c("5000", "10000", "50000", "100000"))) %>%
  ggplot(., aes(x = "n50", y = Bin, fill = n50)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_viridis_d(option = "G", direction = -1, name = "N50", drop = FALSE)
  
n50
  
```


### Completeness
```{r}
completeness <- ggplot(methanobrevibacter_stats, aes(x = "checkM.completeness", y = Bin, fill = checkM.completeness)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_gradient(low = "white", high = "black", na.value="gray", name = "Completeness", limits = c(50, 100))

completeness
```

### Total SNPs
```{r}
total_SNPs <- ggplot(fastBAPS, aes(x = "total_snps", y = Bin, fill = total_snps)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_viridis_c(option = "A", name = "Total SNPs", limits = c(4, 3520))

total_SNPs
```

### Gene Count
```{r}
scc_gene_list <- fread("04-analysis/phylogenies/methanobrevibacter/cluster_4_w_CSC/scc_non_recomb/methanobrevibacter_C4_w_CSC_non_recomb_scc_gene_list.tsv") %>%
  pull()

panaroo <- fread("04-analysis/panaroo/methanobrevibacter/w_new_mag/cluster_4/results_cluster_4/gene_presence_absence.Rtab") %>%
  filter(Gene %in% scc_gene_list) %>%
  pivot_longer(-Gene, values_to = "counts", names_to = "Bin") %>%
  pivot_wider(names_from = "Gene", values_from = "counts") %>%
  rowwise() %>%
  mutate(total_counts = sum(c_across(-Bin), na.rm = TRUE)) %>%
  ungroup()  %>%
  dplyr::select(Bin, total_counts) %>%
  mutate(Bin = if_else(str_detect(Bin, "GC|refbinned"), Bin, 
                        str_replace_all(Bin, "-metaspades_00", ".bin.r.") %>%
                        str_replace_all("-metaspades_0", ".bin.r.") %>%
                        str_replace_all("-metaspades", ".bin.r.") %>%
                        str_replace_all("-megahit_00", ".bin.r.") %>%
                        str_replace_all("-megahit_0", ".bin.r.") %>%
                        str_replace_all("-megahit_", ".bin.r.") %>%
                        str_replace_all("_", "")))

gene_count <- ggplot(panaroo, aes(x = "total_counts", y = Bin, fill = total_counts)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_viridis_c(option = "A", name = "Gene count", limits = c(238, 907))

gene_count
```


## CRISPR
```{r}
crispr_priority <- c("III-A" = 1, "I-B" = 2, "IV -B" = 3, "II-D" = 4)

crispr_colors <- c("III-A" = "#4590fe",
             "I-B" = "#E794C1",
             "IV-B" = "#8D5D4B",
             "II-D" = "black",
             "None" = "#f0f0f0")


methanobrevibacter_cas <- fread("./04-analysis/cctyper/methanobrevibacter/methanobrevibacter_cctyper_cas_table.tsv") %>%
  rename(source = MAG_source, Bin = sampleID) %>%
  inner_join(., metadata %>%
               select(Bin)) %>%
  filter(str_detect(out_type, "known")) %>%
  dplyr::select(Bin, Best_type) %>%
  distinct()

methanobrevibacter_cas_ranked <- methanobrevibacter_cas %>%
  mutate(priority = crispr_priority[Best_type]) %>%
  arrange(Bin, priority)

crispr_pairs <- methanobrevibacter_cas_ranked %>%
  group_by(Bin) %>%
  summarise(
    types = list(Best_type),
    .groups = "drop"
  ) %>%
  mutate(
    n = map_int(types, length),
    crispr1 = map2_chr(types, n, ~ .x[1]),
    crispr2 = map2_chr(types, n, ~ ifelse(.y > 1, .x[2], "None"))  # only copy if only one element
  ) %>%
  dplyr::select(Bin, crispr1, crispr2) %>%
  full_join(., metadata %>%
              select(Bin)) %>%
  mutate(crispr1 = ifelse(is.na(crispr1), "None", crispr1)) %>%
  mutate(crispr2 = ifelse(is.na(crispr2), "None", crispr2)) 

methanobrevibacter_cas_input <- crispr_pairs %>%
  pivot_longer(cols = c(crispr1, crispr2), names_to = "category", values_to = "crispr_metadata")


heatmap_crispr <- ggplot(methanobrevibacter_cas_input, aes(x = category, y = Bin, fill = crispr_metadata)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank()) +
  scale_fill_manual(
    values = c(crispr_colors, "#f0f0f0"), 
    na.value = "#f0f0f0" 
  )

heatmap_crispr

```



## CAZyme
```{r}
cazymes_methanobrevibacter <-fread("05-results/methanobrevibacter_dbcan3_cazyme_table.tsv")


cazymes_methanobrevibacter <- cazymes_methanobrevibacter %>%
  inner_join(., metadata, by = "Bin")

cazymes_methanobrevibacter_table <- cazymes_methanobrevibacter %>%
mutate(HMMER = str_replace_all(HMMER, "\\([0-9]+-[0-9]+\\)",""),
         HMMER = str_replace_all(HMMER, "_[0-9]+","")) %>%
  mutate(dbCAN_sub = str_replace_all(dbCAN_sub, "_e[0-9]+","")) %>%
  separate_longer_delim(HMMER, delim = "+") %>%  
  filter(`#ofTools` == 3) %>%
  dplyr::select(Bin, HMMER) %>%
  unique() 

cazyme_categories <- fread("./00-documentation/cazyme_functional_categories.tsv")

cazymes_methanobrevibacter_table2 <- cazymes_methanobrevibacter_table %>%
  unique() %>%
  full_join(., cazyme_categories) %>%
  dplyr::select(Bin, HMMER, functional_category)


heatmap_cazyme <- cazymes_methanobrevibacter_table2 %>%
  mutate(functional_category = str_replace_all(functional_category, "Peptidoglycan/cell wall processing", "Cell wall processing")) %>%
  filter(!HMMER == "CE4") %>%
  filter(!HMMER == "GH13") %>%
  full_join(., metadata %>%
              select(Bin)) %>%
  mutate(functional_category = as.character(trimws(functional_category))) %>%
  complete(HMMER, Bin, fill = list(functional_category = NA)) %>%
  filter(!is.na(HMMER)) %>% 
  mutate(HMMER = factor(HMMER, levels = c("GT2", "GT4", "GT66", "GT81", "GT116", "GH154"))) %>%
  mutate(present = 1) %>%  
  ggplot(aes(x = HMMER, y = Bin, fill = functional_category)) +
  geom_tile(color = "white") +
  scale_fill_manual(values = c(cazyme_colors, "#f0f0f0"),
                    na.value = "#f0f0f0") +
  theme_void() +
  theme(
    axis.text.y = element_blank(),    
    axis.ticks.y = element_blank(),
    axis.title.y = element_blank(),
    axis.title.x = element_blank(),     
    panel.grid = element_blank()
  ) +
  guides(fill = guide_legend(title = NULL)) 


heatmap_cazyme

#GT2, GT4, GT66, GT81, GT116, GH154 
```




## nimI (presence/absence)
```{r}
nimI <- fread("./04-analysis/card/methanobrevibacter/methanobrevibacter_card_table.tsv") %>%
  dplyr::select(Bin, Best_Hit_ARO) %>%
  filter(str_detect(Best_Hit_ARO, "nimI")) %>%
  unique() %>%
  mutate(nimI = "Present") %>%
  full_join(., metadata) %>%
  mutate(nimI = if_else(is.na(nimI), "Absent", nimI)) %>%
  filter(primary_cluster == "4")

heatmap_nimI <- ggplot(nimI, aes(x = "nimI", y = Bin, fill = nimI)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_manual(
    values = c("Absent" = "#fbf1f6", "Present" = "#97587bff"),
    na.value = "white")

heatmap_nimI

```

## Prophage
```{r}
methanobrevibacter_prophages <- fread("../cmc_deep_assembly/methanobrevibacter_prophages.tsv") %>%
  filter(!str_detect(checkv_quality, "Low-quality")) %>%
  dplyr::select(Bin) %>%
  unique() %>%
  mutate(Prophage = "Present") %>%
  full_join(., metadata) %>%
  mutate(Prophage = replace_na(Prophage, "Absent")) %>%
  inner_join(., metadata %>%
               select(Bin))

prophage_colors <- c("Present" = "#f6865e",
              "Absent" = "#FBD9D2")

prophage_heatmap <- ggplot(methanobrevibacter_prophages, aes(x = "Prophage", y = Bin, fill = Prophage)) +
  geom_tile() +
  theme_void() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_manual(
    values = c("Absent" = "#FBD9D2", "Present" = "#f6865e"),
    na.value = "white")

prophage_heatmap
```

## Methanogenesis

```{r}
methanogenesis_pres_abs <- fread("05-results/methanobrevibacter_methanogenesis_genes_presence_absence.tsv")

gene_counts <- methanogenesis_pres_abs %>%
  dplyr::select(-Gene) %>%
  summarise(across(everything(), \(x) sum(x, na.rm = TRUE))) %>%
  pivot_longer(cols = everything(), names_to = "Bin", values_to = "total") %>%
  mutate(Bin = if_else(str_detect(Bin, "GC|refbinned"), Bin, 
                        str_replace_all(Bin, "-metaspades_00", ".bin.r.") %>%
                        str_replace_all("-metaspades_0", ".bin.r.") %>%
                        str_replace_all("-metaspades", ".bin.r.") %>%
                        str_replace_all("-megahit_00", ".bin.r.") %>%
                        str_replace_all("-megahit_0", ".bin.r.") %>%
                        str_replace_all("-megahit_", ".bin.r.") %>%
                        str_replace_all("_", ""))) %>%
  full_join(., metadata %>%
              select(Bin, primary_cluster)) %>%
  filter(primary_cluster == "4") %>%
  unique()
            
gene_counts_bar_plot <- gene_counts %>%
  ggplot(., aes(x = total, y = reorder(Bin, total))) +
    geom_col(fill = "#C7E9C0") +
    theme_classic() +
    theme(
      axis.title.x = element_blank(),
      axis.title.y = element_blank(),
      axis.text.y = element_blank(),     
      axis.ticks.y = element_blank(),    
      axis.line.y = element_blank(),
      #axis.ticks.x = element_blank(),
      #axis.line.x = element_blank(),
      #axis.text.x = element_blank(),
      axis.text.x = element_text(size = 5),
      plot.background = element_rect(fill = "white", color = NA)
    )
  
  

gene_counts_bar_plot

```

## Panaroo
```{r}
heatmap_panaroo_data <- fread("./04-analysis/panaroo/methanobrevibacter/w_new_mag/cluster_4/results_cluster_4/gene_presence_absence.Rtab") %>%
  pivot_longer(!Gene, names_to = "Bin", values_to = "Gene presence/absence") %>%
  pivot_wider(names_from = "Gene", values_from = "Gene presence/absence") %>%
    mutate(Bin = if_else(str_detect(Bin, "GC|refbinned"), Bin, 
                        str_replace_all(Bin, "-metaspades_00", ".bin.r.") %>%
                        str_replace_all("-metaspades_0", ".bin.r.") %>%
                        str_replace_all("-metaspades", ".bin.r.") %>%
                        str_replace_all("-megahit_00", ".bin.r.") %>%
                        str_replace_all("-megahit_0", ".bin.r.") %>%
                        str_replace_all("-megahit_", ".bin.r.") %>%
                        str_replace_all("_", ""))) %>%
  pivot_longer(!Bin, names_to = "gene", values_to = "Gene presence/absence")

#rank the genes by abundance for clustering on the heatmap
gene_abundance <- fread("./04-analysis/panaroo/methanobrevibacter/w_new_mag/cluster_4/results_cluster_4/gene_presence_absence.Rtab") %>%
  adorn_totals(where = "col") %>%
  dplyr::select(Gene, Total) %>%
  arrange(desc(Total)) %>%
  dplyr::select(Gene) %>%
  pull()

heatmap_panaroo <- ggplot(heatmap_panaroo_data, aes(x = factor(gene, levels=gene_abundance), y = Bin, fill = `Gene presence/absence`)) +
  geom_tile() +
  theme_minimal() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  scale_fill_gradient(low = "#F0F8FF", high = "#132157", na.value="white")

heatmap_panaroo
```



## Combine 
```{r}
methanobrevibacter_full_tree_C4 <- mag_source_heatmap %>% insert_left(methanobrevibacter_tree_c4_age, width = 6) %>% insert_right(heatmap_age, width = 1) %>% insert_right(continent_heatmap) %>% insert_right(latitude, width = 1) %>% insert_right(longitude, width = 1) %>% insert_right(species_clusters_heatmap) %>% insert_right(completeness, width = 1) %>% insert_right(contigs, width = 1) %>% insert_right(n50, width = 1) %>% insert_right(total_SNPs, width = 1) %>% insert_right(gene_count, width = 1) %>% insert_right(heatmap_crispr, width = 1) %>% insert_right(prophage_heatmap, width = 1) %>% insert_right(heatmap_nimI, width = 1) %>% insert_right(heatmap_cazyme, width = 1) %>% insert_right(gene_counts_bar_plot, width = 1) %>% insert_right(heatmap_panaroo, width = 1)

methanobrevibacter_full_tree_C4

#ggsave("./05-results/methanobrevibacter_cluster4_full_tree_tmp.png", plot = methanobrevibacter_full_tree_C4, device = "png", scale = 1, width = 26, height = 7, units = c("in"), dpi = 300) 
```


Save another version with the scale bars
```{r}

gene_counts_bar_plot_w_scales <- gene_counts %>%
  ggplot(., aes(x = total, y = reorder(Bin, total))) +
    geom_col(fill = "#C7E9C0") +
    theme_classic() +
    theme(
      axis.title.x = element_blank(),
      axis.title.y = element_blank(),
      axis.text.y = element_blank(),     
      axis.ticks.y = element_blank(),    
      axis.line.y = element_blank(),
      #axis.ticks.x = element_blank(),
      #axis.line.x = element_blank(),
      #axis.text.x = element_blank(),
      axis.text.x = element_text(size = 5),
      plot.background = element_rect(fill = "white", color = NA)
    )

```


```{r}
methanobrevibacter_full_tree_w_scales <- mag_source_heatmap %>% insert_left(methanobrevibacter_tree_age, width = 6) %>% insert_right(heatmap_age, width = 1) %>% insert_right(continent_heatmap) %>% insert_right(latitude, width = 1) %>% insert_right(longitude, width = 1) %>% insert_right(species_clusters_heatmap) %>% insert_right(completeness, width = 1) %>% insert_right(contigs, width = 1) %>% insert_right(n50, width = 1) %>% insert_right(total_SNPs, width = 1) %>% insert_right(gene_count, width = 1) %>% insert_right(heatmap_crispr, width = 1) %>% insert_right(prophage_heatmap, width = 1) %>% insert_right(heatmap_nimI, width = 1) %>% insert_right(heatmap_cazyme, width = 1) %>% insert_right(gene_counts_bar_plot_w_scales, width = 1)

methanobrevibacter_full_tree_w_scales

#ggsave("./05-results/methanobrevibacter_cluster4_full_tree_w_scales.pdf", plot = methanobrevibacter_full_tree_w_scales, device = "pdf", scale = 1, width = 25, height = 7, units = c("in"), dpi = 300) 
```


